File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> numbers, int target) {
    int answer = 0;
    stack<pair<int, int>> dfs;
    dfs.push({0, 0});
    
    while(!dfs.empty()){
        int index = dfs.top().first;
        int value = dfs.top().second;
        dfs.pop();
        if(index == numbers.size()){
            if(value == target)answer++;
        }else{
            dfs.push({index + 1, value + numbers[index]});
            dfs.push({index + 1, value - numbers[index]});
        }
    }
    
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        Stack<int[]> dfs = new Stack<>();
        dfs.push(new int[]{0, 0}); // {index, value}
        
        while (!dfs.isEmpty()) {
            int[] current = dfs.pop();
            int index = current[0];
            int value = current[1];
            
            if (index == numbers.length) {
                if (value == target) {
                    answer++;
                }
            } else {
                dfs.push(new int[]{index + 1, value + numbers[index]});
                dfs.push(new int[]{index + 1, value - numbers[index]});
            }
        }
        
        return answer;
    }
}